Source/Supervisor/Supervisor.cpp

```diff
@@ -51,6 +49,32 @@ extern "C"
 
         return ret;
     }
+    
+    // Return a pointer to the SupervisorDeviceInstance for the given index
+    const SupervisorDeviceInstance_t* SupervisorIdx2Inst(uint32_t idx)
+    {
+        if(idx >= Supervisor::DeviceVector.size()) return PNULL;    // OOR
+        
+        return &(Supervisor::DeviceVector[idx]);
+    }
+    
+    // Return the name of the given index
+    std::string SupervisorIdx2Name(uint32_t idx)
+    {
+        if(idx >= Supervisor::DeviceVector.size())
+        {
+            //out of range Idx
+            return "";
+        }
+        
+        return Supervisor::DeviceVector[idx].Name;
+    }
```

Comment:
If I'm understanding correctly, both of these out-of-range outputs (L56 and L67) are checked by the caller, propagating errors appropriately. 🍠 just to make sure.

Reply:
Yes, that is the intention.